### DISCUSSION TITLE
Windows Support?

### DISCUSSION THREAD

**Discussion by Tuxdiver** [choria-io/general#38] [General]
Hello,

is there a windows version of choria? 

There was a hint in the april 2020 release:
"**We have done major work on Windows support - it can run as a service, write logs to Windows event log and we have initial packages.**"

I tried to "include mcollective" as we do on ubuntu, but it fails - so I think, windows is still not supported?

Greetings,
Dirk.

**Comment by ripienaar** [comment]
There is a msi that you have to install on your own since windows (without chocolatey etc) doesn’t really have a yum or apt concept 

a few releases ago I published a msi. 

Not many windows users so it’s not been a priority to go through the pain of building that for every release.

**Comment by Tuxdiver** [comment]
Ah - I found Choria-0.25.1.msi and installed it on my windows machine and it works - ubuntu mco now sees the windows machine.

I had to change a few parameters:
```
# no windows repo
choria::manage_package_repo: false

# the next two were not needed for linux but are for windows
choria::manage_service: true
choria::server_service_enable: true

# we had this setting for linux, so change it for windows or do not configure it at all, defaults are ok:
mcollective_agent_puppet::server_config:
  command: "C:/Program Files/Puppet Labs/Puppet/bin/puppet.bat agent"
```

What is missing at the moment is the installation of the choria.msi - but I think I will find a way...


Thanks a lot for your help and for your work, of course!

**Comment by ripienaar** [comment]
Cool, yeah I cant do a standard thing for the msi you pretty much gotta figure that out.  glad you got it sorted.  Keen to get some help to move my tests for windows to GHA and also maybe do automation to build a msi during release.
